ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -1083,21 +1164,21 @@ public void replicaThreadTest() throws Exception {
           .isEqual(partitionId.toString()) ? 3 : 0;
     }
     currentTimeMs = time.milliseconds();
-    replicaThread.replicate(replicasToReplicateList);
+    replicaThread.replicate();
     assertEquals("Replication for all replicas should be disabled and the thread should sleep",
         currentTimeMs + config.replicationReplicaThreadIdleSleepDurationMs, time.milliseconds());
     assertMissingKeys(missingKeys, batchSize, replicaThread, remoteHost, replicasToReplicate);
 
     // 3. forward the time and run replicate and verify the replication.
     time.sleep(config.replicationSyncedReplicaBackoffDurationMs);
-    replicaThread.replicate(replicasToReplicateList);
+    replicaThread.replicate();
     missingKeys = new int[replicasToReplicate.get(remoteHost.dataNodeId).size()];
     assertMissingKeys(missingKeys, batchSize, replicaThread, remoteHost, replicasToReplicate);
 
     // 4. add more put messages and verify that replication continues and is throttled appropriately.
     addPutMessagesToReplicasOfPartition(partitionId, Arrays.asList(localHost, remoteHost), 3);
     currentTimeMs = time.milliseconds();
-    replicaThread.replicate(new ArrayList<>(replicasToReplicate.values()));
+    replicaThread.replicate();
     assertMissingKeys(missingKeys, batchSize, replicaThread, remoteHost, replicasToReplicate);
     assertEquals("Replica thread should sleep exactly " + expectedThrottleDurationMs + " since remote has new token",
```

Comment:
It's generally not a good idea to rely on times being exact, especially to the ms.

Reply:
`MockTime` is used.